src/fsharp/SyntaxTree.fs

```diff
@@ -139,13 +139,13 @@ type SynConst =
     | UserNum of value: string * suffix: string
 
     /// F# syntax: verbatim or regular string, e.g. "abc"
-    | String of text: string * range: range
+    | String of text: string * isVerbatim: bool * isTripleQuote: bool * range: range
```

Comment:
Instead bools, should we create a `SynStringKind`?
```
[<Struct;RequireQualifiedAccess>]
type SynStringKind =
    | Regular
    | Verbatim
    | TripleQuote
```
Personally, I'm not a fan of using `bool`s to represent the kinds of things. Though, in this case, I don't think there is a scenario where `isVerbatim` and `isTripleQuote` could both be `true`; we should make that un-representable.

For the Bytes case, it could also be:
```
[<Struct;RequireQualifiedAccess>]
type SynByteStringKind =
    | Regular
    | Verbatim
```

Reply:
Hi @TIHan, great suggestion. I tried to solve this in the parsing as the lex finisher is used by both types.